src/server/transaction.cc

```diff
@@ -929,6 +943,12 @@ bool Transaction::ScheduleUniqueShard(EngineShard* shard) {
   DCHECK(shard_data_.size() == 1u || multi_->mode == NON_ATOMIC);
   DCHECK_NE(unique_shard_id_, kInvalidSid);
 
+  // callback with no args that needs to be scheduled can run ooo, no keys need to be locked.
+  if (empty_args_) {
+    RunQuickie(shard);
+    return true;
+  }
+
```

Comment:
Not sure we need this at all, because it doesn't have any keys either way and the lock check will pass.

Reply:
The other question is whether we should respect global locks - and it looks like we should.

For example, in-between switching journal subscribers from full to stable sync, we issue a global tx to block all journal events. Now this event might get simply lost for some replica if its fired during this window

Reply:
There are many CHECK fails on this flow this is why I added this, but true we do need to respect the gloabal locks, I will fix this